src/main/java/org/javarosa/core/util/ArrayUtilities.java

```diff
@@ -33,6 +35,20 @@ public static boolean arraysEqual(byte[] array1, byte[] array2) {
     }
 
     /**
+     * Reverses an Array
+     */
+    public static <T> T[] reverse(Object[] myArray, Class<? extends T[]> tClass) {
```

Comment:
@wpride It seems like currently we are not using 'commons-lang3' library in the project. Should I go ahead and add it.